server/agent/core/builder.py

```diff
@@ -23,14 +23,13 @@ def _build_agent(
         self,
         tools: list[BaseTool],
         llm: BaseLanguageModel,
-        prompt: ChatPromptTemplateConfig,
         callback_handler: BaseCallbackHandler,
     ) -> BaseAgent:
         return self._config.agent_class(
             tools=tools,
             llm=llm,
             llm_registry=self._llm_registry,
-            prompt=prompt,
+            prompt=self._prompt,
```

Comment:
I don't think this should be a part of this PR, same for the type changes above

Reply:
Changed, thanks!